tests/providers/microsoft/azure/hooks/test_asb_admin_client.py

```diff
@@ -0,0 +1,104 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import json
+from unittest import mock
+
+import pytest
+from azure.servicebus.management import ServiceBusAdministrationClient
+
+from airflow import AirflowException
+from airflow.models import Connection
+from airflow.providers.microsoft.azure.hooks.asb_admin_client import AzureServiceBusAdminClientHook
+
+
+class TestAzureServiceBusAdminClientHook:
+    def setup_class(self) -> None:
+        self.queue_name: str = "test_queue"
+        self.conn_id: str = 'azure_service_bus_default'
+        self.connection_string = (
+            "Endpoint=sb://test-service-bus-provider.servicebus.windows.net/;"
+            "SharedAccessKeyName=Test;SharedAccessKey=1234566acbc"
+        )
+        self.client_id = "test_client_id"
+        self.secret_key = "test_client_secret"
+        self.mock_conn = Connection(
+            conn_id='azure_service_bus_default',
+            conn_type='azure_service_bus',
+            login=self.client_id,
+            password=self.secret_key,
+            extra=json.dumps({"connection_string": self.connection_string}),
+        )
+
+    @mock.patch(
+        "airflow.providers.microsoft.azure.hooks.asb_admin_client."
+        "AzureServiceBusAdminClientHook.get_connection"
+    )
+    def test_get_conn(self, mock_connection):
+        mock_connection.return_value = self.mock_conn
+        hook = AzureServiceBusAdminClientHook(azure_service_bus_conn_id=self.conn_id)
+        assert isinstance(hook.get_conn(), ServiceBusAdministrationClient)
+
+    @mock.patch('azure.servicebus.management.QueueProperties')
+    @mock.patch(
+        'airflow.providers.microsoft.azure.hooks.asb_admin_client.AzureServiceBusAdminClientHook.get_conn'
+    )
+    def test_create_queue(self, mock_sb_admin_client, mock_queue_properties):
+        """
+        Test `create_queue` hook function with mocking connection, queue properties value and
+        the azure service bus `create_queue` function
+        """
+        mock_queue_properties.name = self.queue_name
```

Comment:
Do you need to mock QueueProperties? can't just you initialise like
```
QueueProperties(name="queue_name")
```

Reply:
Yes it has to be mocked